assets/src/components/map.tsx

```diff
@@ -476,10 +478,15 @@ export const FollowerStatusClasses = (
 
 export const MapFollowingPrimaryVehicles = (props: Props) => {
   const state = useInteractiveFollowerState(),
-    { shouldFollow } = state
+    { shouldFollow, setShouldFollow } = state
 
   const positions: LatLng[] = props.vehicles.map(vehicleToLeafletLatLng)
 
+  useEffect(
+    () => setShouldFollow(true),
+    [props.followerResetKey, setShouldFollow]
```

Comment:
I messed up by adding this as a commit-level comment rather than PR comment - copying here for easier conversation! Original discussion [here](https://github.com/mbta/skate/commit/4877dbec0456966aa20ada968e3e4258183cf793#r102950878):

> One thing I'm trying to understand is why this fix is needed for the shuttle page and not the search map page which has a similar sidebar selection mechanism. It looks like MapDisplay avoids this issue by using BaseMap with a custom child that uses some [custom selection state](https://github.com/mbta/skate/blob/e1bb5da335693db2ed0f38238f13a4861d2b5a93/assets/src/components/mapPage/mapDisplay.tsx#L331).
> 
> I'd be interested in seeing if a similar solution would work here (if-minor, which I suspect it might not be!) If it isn't minor, what do you think about separating the <BaseMap> props type from the <MapFollowingPrimaryVehicles/> props type so that this prop is only added for <MapFollowingPrimaryVehicles/>?

Reply:
I was having trouble wording it but I support what Kayla's getting at above. I think the way to go is using `<BaseMap>` on the shuttle page and keeping special logic there rather than adding more props to the `<BaseMap>`

Reply:
Okay, I think I see the advantages but I'm just struggling to tease apart the different pieces of work that would be needed to do this neatly. One thing that jump out to me is that the `<Map />` component actually encapsulates the logic used by the minimap and search page, whereas `<BaseMap />` is the more generic version, and I think it would be good to reorganize and rename some components to make that clearer (so that `<Map />` is the most generic version that all other components build off of).

Reply:
This seems OK to me! I'm struggling a bit with the name `MapFollowingSelectionKey` since the map is still following the primary vehicles, and only the focus is reset based on the selectionKey. Maybe something like `MapWithFocusResetKey`? It still seems ideal to not introduce a new special map if possible, but I consider that non-blocking unless @firestack sees it differently!

Reply:
Do you mean special map, or special prop? I could just use `<Map />` directly from the `<SearchPage />` component and put the logic there, but then it wouldn't be reusable by anything else that has similar functionality where changes in selection are supposed to reset the follower state. That means no new component and no new prop but it's also less modular, and this way the logic isn't added to the same map that other components use.

Reply:
I meant special map, but your point about reusability is well taken and am fully persuaded by it! It would be nice for consistency if `<SearchPage/> ,`<ShuttleMapPage/>`, and any other future interactive map to use the same underlying reusable map component.